### PR TITLE
Populate Source JSON from CAN helper

### PR DESCRIPTION
**Pull request by TomGoBravo**
This PR is the bulk of  https://trello.com/c/hSJN82y9/927-pipe-through-source-names-to-fe

* Creates Source JSON in `CanScraperBase.make_dataset()` instead of provenance and source_url
* Factors out code that makes a DataFrame containing a bunch of Source JSON to `taglib.Source.rename_and_make_tag_df`

## Tested

Ran `./run.py data update` then used `git difftool --no-prompt -t vimdiff main -- data/multiregion-wide-dates.csv` to see `name` was added to the JSON in the CSV in places that already had a URL.

### CODE REVIEW

**Comment by TomGoBravo** [comment]
[Pipe through Source Names to FE](https://trello.com/c/hSJN82y9/927-pipe-through-source-names-to-fe)

**Comment by TomGoBravo** [comment]
`pytest -vv 'tests/cli/data_cli_test.py::test_population_filter'` passes on my linux laptop but failed twice on our server with a vague `node down: Not properly terminated` and no obvious stack :-(

File: libs/datasets/taglib.py [hunk 1]
```diff
@@ -141,6 +141,36 @@ class Source(TagInTimeseries):
 
     TAG_TYPE = TagType.SOURCE
 
+    @staticmethod
+    def rename_and_make_tag_df(
+        in_df: pd.DataFrame, *, source_type=None, rename: Mapping[str, str]
```
Comment (ghop02) [inline libs/datasets/taglib.py:146]: does source_type have a clear type?

**Review by ghop02 — APPROVED** [review]
LGTM!

File: libs/datasets/taglib.py (hunk 1 above)
Comment (TomGoBravo) [reply libs/datasets/taglib.py]: added in 99a1e4d

**Comment by TomGoBravo** [comment]
Throwing in towel on broken test and created https://trello.com/c/hljeZmmD/996-fix-testpopulationfilter-currently-marked-as-skip

**Pull request merged**